contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,202 @@
+accepted.
+
+The schema is copied from
+[#132](https://github.com/kubernetes/community/pull/132) with a few
+modifications.
+
+```golang
+type AdmissionControlConfiguration struct {
+    TypeMeta // although this object could simply be serialized like ComponentConfig
+
+    // ResourceInitializers is a list of resources and their default initializers
+    ResourceInitializers []ResourceDefaultInitializer
+
+    ExternalAdmissionHooks []ExternalAdmissionHook
+}
+
+// Because the order of initializers matters, and each resource might need
+// differnt order, the ResourceDefaultInitializers are indexed by Resource.
+type ResourceDefaultInitializer struct {
+    // Resource identifies the type of resource to be initialized that should be
+    // initialized
+    Resource GroupResource
+    // Initializers are the default names that will be registered to this resource
+    Initializers []Initializer
+}
+
+type Initializer struct {
+    // Name is the string that will be registered to the resource that needs
+    // initialization.
+    Name string
+
+    // **Optional for alpha implement**
+    // FailurePolicy defines what happens if there is no initializer controller
+    // takes action. Allowed values are Ignore, or Fail. If "Ignore" is set, 
+    // apiserver removes initilizer from the initializers list of the resource
+    // if the timeout is reached; If "Fail" is set, apiserver returns timeout
+    // error if the timeout is reached.
+    FailurePolicy FailurePolicyType
+
+    // **Optional for alpha implement**
+    // If timeout is reached, the intializer is removed from the resource's
+    // initializer list by the apiserver.
+    // Default to XXX seconds.
+    Timeout *int64
+}
+
+type ExternalAdmissionHook struct {
+    // Operations is the list of operations this hook will be invoked on - Create, Update, or *
+    // for all operations. Defaults to '*'.
+    Operations []string
+    // Resources are the resources this hook should be invoked on. '*' is all resources.
+    Resources []string
+    // Subresources are the list of subresources this hook should be invoked on. '*' is all resources.
+    Subresources []string
+
+    // ClientConfig defines how to talk to the hook.
+    ClientConfig AdmissionHookClientConfig
+
+    // FailurePolicy defines how unrecognized errors from the admission endpoint are handled -
+    // allowed values are Ignore, Retry, Fail. Default value is Fail
+    FailurePolicy FailurePolicyType
+}
+
+// AdmissionHookClientConfig contains the information to make a TLS
+// connection with the webhook
+// **very similar to the schema of kubeconfig**
+type AdmissionHookClientConfig struct {
+	// Address of the external admission hook, could be a host string, 
+    // a host:port pair, or a URL.
+	Address string
+	// ClientCertificate is the path to a client cert file for TLS.
+	ClientCertificate string
+    // ClientCertificateData contains PEM-encoded data from a client cert file
```

Comment:
A cert key is sensitive data. I think it kinda breaks the model to pass the key around like this.